### PR TITLE
[370] implement a flight check for personal email addresses used for referees

### PR DESCRIPTION
**Pull request by elceebee**
## Context

Although our guidance does not ban the use of personal email addresses (eg [gmail.com](http://gmail.com/)) for references, many providers will reject an application that contains them, or contact the applicant to request a professional alternative.

The aim of this ticket is to add a flight check screen, shown when adding a reference using a personal email address, both when it is entered, and before submitting.

There are four places we are interrupting: 
- When creating / updating email addresses on the application form. This can only happen before any applications have been submitted
- When updating email addresses when accepting an offer. When people accept offers, they are given the opportunity to confirm their references. 
- When creating a new reference after offer. Candidate are able to add references when an offer after an offer has been accepted
- When submitting an application for the first time -- if they haven't submitted anything yet, we ask them to go back to the application form and update the personal email addresses. 

## Changes proposed in this pull request

Two different kinds of interruption pages
1. When someone is creating / adding a personal email address
2. When someone is submitting an application and one or more of their reference email addresses is a personal email address
3. Lots of controllers / routes / tests for the scenarios described in the context section
4. A method for determining if the email address is personal

| creating and adding references | submitting for the first time | when accepting an offer | after an offer is accepted |
| --- | --- | --- | --- |
| https://github.com/user-attachments/assets/85340e3a-07b4-473e-ba1f-512dfafc1586 | https://github.com/user-attachments/assets/2c74ac44-9218-48e2-9013-c40a18041d4b |https://github.com/user-attachments/assets/97c782e5-cb62-4d72-bcd7-a66cbe693ca9 | https://github.com/user-attachments/assets/b6a0f963-e6f2-41fa-a090-b227c0532664 |

## Guidance to review

- Can you think of any other places where we add / edit / confirm references? 
- Check back links and navigation really thoroughly
- Any better ideas for a list of personal email address domains? This was the list that was in the ticket as a suggestion (presumably by david), but if you know of anything better, let me know. 

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)
- [ ] Inform data insights team due to database changes
- [x] Make sure all information from the Trello card is in here
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally
- [x] Add PR link to Trello card

### CODE REVIEW

**Comment by elceebee** [comment]
[Implement a flight check for personal email addresses used for referees](https://trello.com/c/UTXpQNsH/370-implement-a-flight-check-for-personal-email-addresses-used-for-referees)

File: app/controllers/candidate_interface/course_choices/review_references_interruption_controller.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module CandidateInterface
+  module CourseChoices
+    class ReviewReferencesInterruptionController < CandidateInterface::CourseChoices::BaseController
+      before_action :redirect_to_your_applications_if_submitted
+
+      def show
+        @application_choice = current_application.application_choices.find(params[:application_choice_id])
```
Comment (CatalinVoineag) [inline app/controllers/candidate_interface/course_choices/review_references_interruption_controller.rb:7]: can we do `params.require(:application_choice_id) here?`

**Review by CatalinVoineag — APPROVED** [review]
Tested it on review, looks good and the back links work, nice work 👍🏻

**Review by dcyoung-dev — APPROVED** [review]
The `next_path` methods in the controllers feel like they could be extracted into a testable model - I'm not saying we should do it but I do wonder if it'd make testing a bit more robust, a nice unit test would be cool

**Review by PeteWilliams — APPROVED** [review]
All good from a design perspective.

**Pull request merged**